Assets/Scripts/CobwebInteraction.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+
+public class CobwebInteraction : MonoBehaviour
+{
+    [SerializeField] private float interactionTime = 3;
+    private float interactionTimeLeft;
+
+    [SerializeField] private GameObject progressBar;
+    [SerializeField] private GameObject textElement;
+
+    [SerializeField] private float interactionDistance;
+    private GameObject cobweb;
+
+    private Slider slider;
+    private bool taskCompleted = false;
+
+    //[SerializeField] private AudioClip endTaskClip;
+    //private AudioScript audioPlayer;
+
+    void Start()
+    {
+        interactionTimeLeft = interactionTime;
+        slider = progressBar.GetComponent<Slider>();
+        slider.maxValue = interactionTime;
+        //audioPlayer = GetComponentInParent<AudioScript>();
+    }
+
+    void FixedUpdate()
+    {
+        // Bit shift the index of the layer (10) to get a bit mask
+        int layerMask = 1 << 10;
+
+        RaycastHit hit;
+
+        if(interactionTimeLeft > 0)
+        { 
+            // Does the ray intersect any objects excluding the beehive layer
+            if (Physics.Raycast(transform.position, transform.TransformDirection(Vector3.forward), out hit, interactionDistance, layerMask))
+            {
+                textElement.GetComponent<Text>().text = "Press E to break cobweb";
+                cobweb = hit.transform.gameObject;
+                progressBar.SetActive(true);
+                textElement.SetActive(true);
```

Comment:
here you then would change it's activeness by e.g. `progressBar.gameObject.SetActive(false)`